pkg/daemon/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,189 @@
+		},
+		{
+			name:             "ConfigMap managed service doesn't exist on node",
+			existingServices: map[string]*fake.FakeService{},
+			expectedServices: map[string]struct{}{},
+			preserveNode:     false,
+			servicesCMExists: true,
+			configMapServices: []servicescm.Service{
+				{Name: "test1", Dependencies: nil, Priority: 0},
+			},
+		},
+		{
+			name: "Single service",
+			existingServices: map[string]*fake.FakeService{
+				"test1": newTestService("test1", []string{}, true),
+			},
+			expectedServices: map[string]struct{}{},
+			preserveNode:     false,
+			servicesCMExists: true,
+			configMapServices: []servicescm.Service{
+				{Name: "test1", Dependencies: nil, Priority: 0},
+			},
+		},
+		{
+			name: "Upgrade scenario with multiple services",
+			existingServices: map[string]*fake.FakeService{
+				"test1":           newTestService("test1", []string{}, true),
+				"test2":           newTestService("test2", []string{}, true),
+				"test3-unmanaged": newTestService("test3-unmanaged", []string{}, false),
+			},
+			expectedServices: map[string]struct{}{"test3-unmanaged": {}},
+			preserveNode:     true,
+			servicesCMExists: true,
+			configMapServices: []servicescm.Service{
+				{Name: "test1", Dependencies: nil, Priority: 0},
+				{Name: "test2", Dependencies: nil, Priority: 0},
+			},
+		},
+		{
+			name: "Multiple services with only managed dependencies",
+			existingServices: map[string]*fake.FakeService{
+				"test1":           newTestService("test1", []string{"test2", "test4"}, true),
+				"test2":           newTestService("test2", []string{}, true),
+				"test3":           newTestService("test3", []string{"test4"}, true),
+				"test4":           newTestService("test4", []string{}, true),
+				"test5-unmanaged": newTestService("test5-unmanaged", []string{}, false),
+			},
+			expectedServices: map[string]struct{}{"test2": {}},
+			preserveNode:     false,
+			servicesCMExists: true,
+			configMapServices: []servicescm.Service{
+				{Name: "test1", Dependencies: []string{"test2", "test4"}, Priority: 3},
+				{Name: "test3", Dependencies: []string{"test4"}, Priority: 1},
+				{Name: "test4", Dependencies: nil, Priority: 0},
+				{Name: "test5-unmanaged", Dependencies: nil, Priority: 0},
+			},
+		},
+		{
+			name: "Multiple services including unmanaged dependencies",
+			existingServices: map[string]*fake.FakeService{
+				"test1":           newTestService("test1", []string{"test2", "test3-unmanaged"}, true),
+				"test2":           newTestService("test2", []string{}, true),
+				"test3-unmanaged": newTestService("test3-unmanaged", []string{}, false),
+			},
+			expectedServices: map[string]struct{}{},
+			preserveNode:     false,
+			servicesCMExists: true,
+			configMapServices: []servicescm.Service{
+				{Name: "test1", Dependencies: []string{"test2", "test3-unmanaged"}, Priority: 3},
+				{Name: "test2", Dependencies: []string{}, Priority: 2},
+				{Name: "test3-unmanaged", Dependencies: nil, Priority: 0},
+			},
+		},
```

Comment:
Whats the functional difference between these two test cases?
I don't understand why both are needed. 
I cant find any logic that deal with dependencies and 'unmanaged' services

Reply:
It is a vestige from the out of sync cleanup logic. I'll remove the managed vs unmanaged concept from the test suite.